conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -89,15 +93,15 @@ private static void serviceException(HttpServerExchange exchange, ServiceExcepti
 
     private static void qosException(HttpServerExchange exchange, QosException qosException) {
         qosException.accept(QOS_EXCEPTION_HEADERS).accept(exchange);
-        if (qosExceptionHasAdditionalMetadata(qosException)) {
-            log.info("quality-of-service intervention", qosException);
-        } else {
-            log.debug("quality-of-service intervention", qosException);
+
+        if (qosExceptionHasAdditionalMetadata(qosException) || qosLoggingRateLimiter.tryAcquire()) {
+            log.info("Quality-of-Service error handling request", qosException);
         }
```

Comment:
Let's retain the `debug` functionality, and avoid burning the rate limiter when debug is enabled:
```suggestion
        if (log.isDebugEnabled()) {
            log.debug("Quality-of-Service error handling request", qosException);
        } else if (qosExceptionHasAdditionalMetadata(qosException) || qosLoggingRateLimiter.tryAcquire()) {
            log.info("Quality-of-Service error handling request", qosException);
        }
```

Reply:
It might be helpful to mention in the `info` line that these exceptions are logged at info once per second.

Reply:
I don't think the log line is an appropriate place for this, it just makes it more cumbersome to consume these logs.